Mathlib/CategoryTheory/GuitartExact.lean

```diff
@@ -0,0 +1,217 @@
+
+/-- Given `w : TwoSquare T L R B` and a morphism `g : R.obj X₂ ⟶ B.obj X₃`, this is the
+category `CostructuredArrow (w.structuredArrowDownwards X₂) (StructuredArrow.mk g)`,
+see the constructor `JDownwards.mk` for the data that is involved. -/
+abbrev JDownwards :=
+  CostructuredArrow (w.structuredArrowDownwards X₂) (StructuredArrow.mk g)
+
+section
+
+variable (X₁ : C₁) (a : X₂ ⟶ T.obj X₁) (b : L.obj X₁ ⟶ X₃)
+  (comm : R.map a ≫ w.app X₁ ≫ B.map b = g)
+
+/-- Constructor for objects in `w.JRightwards g`. -/
+abbrev JRightwards.mk : w.JRightwards g :=
+  StructuredArrow.mk (Y := CostructuredArrow.mk b) (CostructuredArrow.homMk a comm)
+
+/-- Constructor for objects in `w.JDownwards g`. -/
+abbrev JDownwards.mk : w.JDownwards g :=
+  CostructuredArrow.mk (Y := StructuredArrow.mk a)
+    (StructuredArrow.homMk b (by simpa using comm))
+
+end
+
+namespace EquivalenceJ
+
+/-- Given `w : TwoSquare T L R B` and a morphism `g : R.obj X₂ ⟶ B.obj X₃`, this is
+the obvious functor `w.JRightwards g ⥤ w.JDownwards g`. -/
+@[simps]
+def functor : w.JRightwards g ⥤ w.JDownwards g where
+  obj f := CostructuredArrow.mk (Y := StructuredArrow.mk f.hom.left)
+      (StructuredArrow.homMk f.right.hom (by simpa using CostructuredArrow.w f.hom))
+  map {f₁ f₂} φ :=
+    CostructuredArrow.homMk (StructuredArrow.homMk φ.right.left
+      (by dsimp; rw [← StructuredArrow.w φ]; rfl))
+      (by ext; exact CostructuredArrow.w φ.right)
+  map_id _ := rfl
+  map_comp _ _ := rfl
+
+/-- Given `w : TwoSquare T L R B` and a morphism `g : R.obj X₂ ⟶ B.obj X₃`, this is
+the obvious functor `w.JDownwards g ⥤ w.JRightwards g`. -/
+@[simps]
+def inverse : w.JDownwards g ⥤ w.JRightwards g where
+  obj f := StructuredArrow.mk (Y := CostructuredArrow.mk f.hom.right)
+      (CostructuredArrow.homMk f.left.hom (by simpa using StructuredArrow.w f.hom))
+  map {f₁ f₂} φ :=
+    StructuredArrow.homMk (CostructuredArrow.homMk φ.left.right
+      (by dsimp; rw [← CostructuredArrow.w φ]; rfl))
+      (by ext; exact StructuredArrow.w φ.left)
+  map_id _ := rfl
+  map_comp _ _ := rfl
+
+end EquivalenceJ
+
+/-- Given `w : TwoSquare T L R B` and a morphism `g : R.obj X₂ ⟶ B.obj X₃`, this is
+the obvious equivalence of categories `w.JRightwards g ≌ w.JDownwards g`. -/
+@[simps functor inverse unitIso counitIso]
+def equivalenceJ : w.JRightwards g ≌ w.JDownwards g where
+  functor := EquivalenceJ.functor w g
+  inverse := EquivalenceJ.inverse w g
+  unitIso := Iso.refl _
+  counitIso := Iso.refl _
+
+lemma isConnected_rightwards_iff_downwards :
+    IsConnected (w.JRightwards g) ↔ IsConnected (w.JDownwards g) :=
+  isConnected_iff_of_equivalence (w.equivalenceJ g)
+
+end
+
+/-- Condition on `w : TwoSquare T L R B` expressing that it is a Guitart exact square.
+See lemmas `guitartExact_iff_isConnected_rightwards`, `guitartExact_iff_isConnected_downwards`,
+`guitartExact_iff_final`, `guitartExact_iff_initial` for various characterizations. -/
```

Comment:
Please spell out at least one of these conditions/characterizations. Also, maybe provide a bit of motivation for this notion (as I think it isn't that well-known), or maybe provide a pointer to more info.